pyerrors/fits.py

```diff
@@ -705,8 +705,19 @@ def _combined_fit(x, y, func, silent=False, **kwargs):
         jacobian = auto_jacobian
         hessian = auto_hessian
 
-    x_all = np.concatenate([np.array(o) for o in x.values()])
-    y_all = np.concatenate([np.array(o) for o in y.values()])
+    key_ls = sorted(list(x.keys()))
+
+    if sorted(list(y.keys())) != key_ls:
+        raise Exception('x and y dictionaries do not contain the same keys.')
+
+    if sorted(list(func.keys())) != key_ls:
+        raise Exception('x and func dictionaries do not contain the same keys.')
+
+    if sorted(list(func.keys())) != sorted(list(y.keys())):
```

Comment:
I think this check is unnecessary. If a=b and a=c then b=c.

Reply:
Yes, I removed it, thanks!